packages/flutter/test/material/checkbox_test.dart

```diff
@@ -603,6 +603,56 @@ void main() {
     expect(box.size, equals(const Size(60, 36)));
   });
 
+  testWidgets('Checkbox stops hover animation when removed from the tree.', (WidgetTester tester) async {
+    const Key checkboxKey = Key('checkbox');
+    bool checkboxVal = true;
+
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.padded),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: StatefulBuilder(
+                builder: (_, StateSetter setState) => Checkbox(
+                  key: checkboxKey,
+                  value: checkboxVal,
+                  onChanged: (bool newValue) => setState(() {checkboxVal = newValue;}),
+                ),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(find.byKey(checkboxKey), findsOneWidget);
+    final Offset checkboxCenter = tester.getCenter(find.byKey(checkboxKey));
+    final TestGesture testGesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await testGesture.moveTo(checkboxCenter);
+
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(milliseconds: 25)); // hover animation duration is 50 ms. It is half-way.
+
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.padded),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: Container(),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await testGesture.removePointer();
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{TargetPlatform.macOS}));
```

Comment:
Is there a reason we're only testing this on macOS?

Reply:
@shihaohong yes, this is redundant. Removed it.